cosmovisor/cmd/cosmovisor/cmd/version.go

```diff
@@ -5,7 +5,7 @@ import (
 )
 
 // Version represents Cosmovisor version value. Set during build
-var Version string
+var Version = "1.0.x"
```

Comment:
Maybe the second option will be the best?

Reply:
The only other option I can think of would be to instruct users to use `-ldflags` when running `go install` and then to instruct users to reinstall using `-ldflags` if the version is not set (`PrintVersion` would report that the version has not been set and instruct users on how to reinstall).

That being said, I think your suggestion to use a make command or a script would be a better user experience than updating the install instructions. Also, I don't think we should manually set it in the case that someone forgets to update it before a release - I think it would be better to have it empty than to have it print the wrong version.

Reply:
(not 100% sure) Is there a way to use https://pkg.go.dev/runtime/debug#ReadBuildInfo at runtime to get the version from go.mod?

Reply:
> (not 100% sure) Is there a way to use https://pkg.go.dev/runtime/debug#ReadBuildInfo at runtime to get the version from go.mod?
> 
> Version is here: https://pkg.go.dev/runtime/debug#Module

I don't think that'll work here. When someone does a `go install <thing>`, go will download the source from this repo (either by tag or HEAD) to the user's computer and build it locally. But the local build won't have the `-ldflags` like they're provided via `Makefile` or `.goreleaser.yml`.

I've been digging around on this for a bit, and the only thing I can find that looks like it'll work, is what this PR is doing (as well as re-tagging it).

We'll probably just have to update the release process to include updating this string's value. It probably wouldn't be too tough to write a shell script for automating some of the release steps including the updating of this string, and possibly wrapping that in a `make` command.

Reply:
> at runtime to get the version from go.mod?

There is no _this_ package version in the go.mod. Also, `_this_` module is not stored in the `BuildInfo.Deps`.

Reply:
@ryanchristo adding `-ldflags` in the install instructions doesn't solve the problem:
* we need to remember to update the instructions (README) before the release to update the flag with new version
* if a user forgets to add `ldflags` then the version will be empty.

Reply:
I don't find any other solution then the one I listed. How about creating a "release" script?

Reply:
A release script sounds like the best option to me.

Reply:
During the meeting we decided to:
1. Crete `prepare-release` make job
2. it will call a script which will do required changes.

Reply:
```suggestion
var Version = "1.0.0"
```

manually bumping this seems fine, we should document this in a contributing or release procedure doc.

Reply:
The suggestion above won't work, we don't want `.0` in `master`.

Reply:
we want `\d.\d.x` in master, but `\d.\d.\d` in release